index.d.ts

```diff
@@ -41,7 +41,7 @@ declare namespace alkali {
     * Returns a variable corresponding to the property of the value of this variable
     * @param key The name of the property
     */
-    property<K extends keyof T>(key: KeyType): Variable<T[K]>
+    property<K extends keyof T>(key: K): Variable<YieldedValue<T[K]>>
```

Comment:
I'm not sure if not using the key type here was intentional but using it allows the nested type to flow through, and we can use YieldedValue to ensure a single level of property type nesting